### PR TITLE
BZ#1544535 - Navigation menu highlighting

### PR DESCRIPTION
**Pull request by chalettu**
This fixes the issue of when you refresh your browser or if you click on a item on the dashboard.  It will highlight the appropriate Nav item on the left nav that corresponds to the correct parent state.
Fix for BZ https://bugzilla.redhat.com/show_bug.cgi?id=1544535

@miq-bot add_label bug
@miq-bot add_label gaprindashvili/yes

### CODE REVIEW

**Comment by chalettu** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by chalettu** [comment]
nav_state
This screenshot shows the fix working and what happens when you refresh the page with the order state selected.

File: client/app/core/navigation/navigation-controller.js [hunk 1]
```diff
@@ -82,11 +82,26 @@ export function NavigationController (Text, Navigation, Session, API_BASE, Shopp
...
     refresh()
+    setMenuItemActive()
     if (ShoppingCart.allowed()) {
       ShoppingCart.reload()
     }
+    $scope.$on('$stateChangeSuccess', () => {
+      setMenuItemActive()
+    })
+  }
+  function setMenuItemActive () {
```
Comment (AllenBW) [inline client/app/core/navigation/navigation-controller.js]: What about putting the logic of this function in the `doCheck` rather than a timeout?

**Comment by AllenBW** [comment]
@chalettu is on pto, so updated the pr as per feedback, same result, less icky timeout 
reload

**Comment by himdel** [comment]
@chalettu One more thing missing..

You're setting active to true, but not unsetting active from anywhere else .. so, in the case when you click something in the dashboard, you end up with this:

evil

**Comment by AllenBW** [comment]
Arg, thats my fault @himdel didn't push up a commit 🤦‍♂️ (this has since been pushed)

**Comment by himdel** [comment]
Aah, fixed indeed, thanks! :)

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit d0f9ec7673f5926c26d1fe704b21434fd2fe8830
Author: Martin Hradil <himdel@seznam.cz>
Date:   Wed Feb 14 15:23:54 2018 +0000

    Merge pull request #1385 from chalettu/active-nav-item
    
    BZ#1544535 - Navigation menu highlighting
    (cherry picked from commit a1d21608fa54aee907536702b65ecda86a19c2aa)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1553480
```

**Pull request merged**